core/invoker/src/main/scala/whisk/core/invoker/Invoker.scala

```diff
@@ -164,11 +164,15 @@ object Invoker {
         zkClient.close()
         assignedId
       }
+    val topicBaseName = "invoker"
+    val topicName = topicBaseName + assignedInvokerId
 
-    val invokerInstance = InvokerInstanceId(assignedInvokerId, invokerName)
+    // Define invoker hostname for the health protocol to distinguish invoker topics in Kafka from the actual container names
+    val invokerHostname = scala.util.Properties.envOrNone("HOSTNAME").filter(name => name != topicName)
+    val invokerInstance = InvokerInstanceId(assignedInvokerId, invokerName, invokerHostname)
```

Comment:
As discussed in slack, let's investigate if we can achieve the wanted functionality by using the already existing `name` Option.